src/esw/cameras.py

```diff
@@ -1,387 +1,232 @@
-                        rospy.logerr(f"Error {type(e)} encountered: {e}")
-                        rospy.logerr(f"Camera {index} capture failed. Stopping stream(s).")
-                        self._close_down_all_streams_of_device(index)
-
-    def _close_down_all_streams_of_device(self, device: int) -> None:
-        """
-        Closes down a video device. Usually called if a device crashes while streaming.
-        Must make sure that the device lock is locked.
-        :param device: The integer ID of the device to be closed.
-        """
-        assert self._device_lock.locked(), "self._device_lock must be locked first."
-        previously_was_video_source = self._video_devices[device].is_streaming()
-        assert (
-            previously_was_video_source
-        ), "_close_down_all_streams_of_device should only be called if a streamed device failed"
-        while len(self._video_devices[device].output_by_endpoint.keys()) != 0:
-            endpoint = list(self._video_devices[device].output_by_endpoint.keys())[0]
-            self._video_devices[device].remove_endpoint(endpoint)
-            service, stream = self._service_streams_by_endpoints[endpoint]
-            self._services[service][stream] = CameraCmd(-1, 0)
-        assert self._video_devices[device].video_source is None, "The video source should be None by now"
-        self._active_devices -= 1
-
-    def _sort_camera_commands_by_resolution_and_error_check(self, camera_commands: List[CameraCmd]) -> CameraCmd:
-        """
-        Sorts a list of camera commands by resolution. Also checks for invalid requested devices and resolutions
-        :param camera_commands: The list of camera commands to be sorted
-        :return: The sorted list of camera commands
-        """
-        # If two or more requests ask for same video source in different resolution,
-        # all requests' resolution get set to the one with the lowest resolution. nlogn + n, but n ~ 4
-
-        assert self._device_lock.locked(), "self._device_lock must be locked first."
-
-        for stream, camera_cmd in enumerate(camera_commands):
-            if camera_cmd.device >= len(self._video_devices):
-                rospy.logerr(f"Request device {camera_cmd.device} invalid. Treating as no device instead.")
-                camera_cmd.device = -1
-            if camera_cmd.resolution >= len(self._list_of_resolution_options):
-                rospy.logerr(
-                    f"Request resolution {camera_cmd.resolution} invalid. Treating as lowest resolution instead."
-                )
-                camera_cmd.resolution = 0
-
-        requests: List[Tuple[int, int, int]] = [
-            (camera_cmd.device, camera_cmd.resolution, stream) for stream, camera_cmd in enumerate(camera_commands)
+
+        primary_ip = rospy.get_param("cameras/ips/primary")
+        secondary_ip = rospy.get_param("cameras/ips/secondary")
+
+        self.ips = [primary_ip, secondary_ip]
```

Comment:
Couldn't you just pull all the ips into one dictionary variable and access the IPs you want as 'primary' or 'secondary'

Reply:
Yes that could probably be smarter